### PR TITLE
Infrastructure: Pack Core projects, move to 2.0.0-preview1, and support local incremental versions

### PR DESCRIPTION
**Pull request by dougbu**
- incremental version fix is part of #178
- copy some settings into WebHooks.msbuild for consistent `$(AssemblyRevision)`
- add target to generate unique `$(BuildNumber)` for command-line builds
- use version.props in Packages.proj

Also fix CodeAnalysis build; was duplicating Debug build

nits:
- remove ??? comments and associated commented-out code (if we find problems, we'll fix them)
- rename version.xml to version.props

### CODE REVIEW

File: WebHooks.msbuild [hunk 1]
```diff
@@ -16,15 +16,28 @@
...
+    <_TwoDigitYear>$([MSBuild]::Subtract($([System.DateTime]::UtcNow.Year), 2000))</_TwoDigitYear>
+    <_ThreeDigitDayOfYear>$([System.DateTime]::UtcNow.DayOfYear.ToString().PadLeft(3, '0'))</_ThreeDigitDayOfYear>
+    <AssemblyRevision>$(_TwoDigitYear)$(_ThreeDigitDayOfYear)</AssemblyRevision>
   </PropertyGroup>
 
   <ItemGroup>
     <SolutionsToBuild Include="WebHooks.sln">
       <BuildInParallel>$(BuildInParallel)</BuildInParallel>
     </SolutionsToBuild>
-    <NugetsToBuild Include="**\*.nuspec"/>
```
Comment (dougbu) [inline WebHooks.msbuild]: Was not passed around at all.

File: WebHooks.msbuild [hunk 2]
```diff
@@ -16,15 +16,28 @@
...
   </ItemGroup>
 
+  <Target Name="GetBuildNumber">
+    <!-- Create second-based build number for local builds. 635556672000000000 is Jan 1, 2015. -->
+    <PropertyGroup>
+      <_SecondBasedTimeStamp>$([System.DateTime]::UtcNow.Subtract($([System.DateTime]::FromBinary(635556672000000000))).TotalSeconds.ToString("F0"))</_SecondBasedTimeStamp>
+      <_SecondBasedTimeStamp>t$([System.Int64]::Parse($(_SecondBasedTimeStamp)).ToString("x9"))</_SecondBasedTimeStamp>
+    </PropertyGroup>
+    <Message Text="/p:BuildNumber=$(_SecondBasedTimeStamp)" Importance="High" />
```
Comment (dougbu) [inline WebHooks.msbuild]: If you want `-t000` packages, just run `.\build.cmd`. But if you want packages that can be installed locally, run `.\build.cmd GetBuildNumber` and copy-n-paste its output after `.\build.cmd BuildPackages`.

File: src/Common/CommonAssemblyInfo.cs [hunk 3]
```diff
@@ -26,8 +26,8 @@
 
 #if ASPNETWEBHOOKS
 #if !BUILD_GENERATED_VERSION
-[assembly: AssemblyVersion("1.1.0.0")]
-[assembly: AssemblyFileVersion("1.1.0.0")]
+[assembly: AssemblyVersion("2.0.0.0")]
+[assembly: AssemblyFileVersion("2.0.0.0")]
```
Comment (dougbu) [inline src/Common/CommonAssemblyInfo.cs]: This file is used in the old-world projects. For now, they won't have the same assembly attributes e.g. no `[assembly: AssemblyInformationalVersion("2.0.0-preview1-t000")]` or `[assembly: AssemblyMetadata("BuildNumber", "t000")]`.

File: version.props [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<Project>
+  <PropertyGroup>
+    <VersionPrefix>2.0.0</VersionPrefix>
```
Comment (dougbu) [inline version.props]: Yup, I copied this from MVC then changed the `$(VersionPrefix)` 😈

File: WebHooks.msbuild [hunk 5]
```diff
@@ -62,25 +75,26 @@
 
   <Target Name="RestorePackages" DependsOnTargets="DownloadNuGet">
     <Message Text="Restoring NuGet packages..." Importance="High" />
-    <Exec Command='"$(NuGetExe)" restore "%(SolutionsToBuild.Identity)" -PackagesDirectory packages -NonInteractive -Verbosity quiet -ConfigFile "$(MsBuildThisFileDirectory)\.nuget\NuGet.Config"' />
+    <Exec Command='SET BuildNumber=$(BuildNumber)
```
Comment (pranavkm) [inline WebHooks.msbuild]: Could these use the restore target? I think it also works on old-type csproj files now (if you have the relevant SDKs installed).

Comment (dougbu) [reply WebHooks.msbuild]: Was thinking about that. I'll test it out&hellip;

If it works, do you want to see the new version before I merge into `dev`?

Comment (dougbu) [reply WebHooks.msbuild]: (Not sure why this comment got hidden as outdated. I only moved an XML comment from a couple of lines down.)

Experiment failed. The `<restore/>` target supports only `<PackageReference />` and `<ProjectReference />` in the project file. https://docs.microsoft.com/en-us/nuget/schema/msbuild-targets#restore-target isn't very explicit about this but other sources are.

**Comment by dougbu** [comment]
FYI :up::date: just moved a comment

**Comment by dougbu** [comment]
1d8c463e95

**Pull request closed without merging**